### PR TITLE
Enhance manifest, fix filename, reduce required permissions

### PR DESCRIPTION
**Pull request by Eagle3386**
- [X] Fixes "{[wxIF's GUID]}.xpi" as filename instead of extension's name
- [x] Sets minimum version to FF48 (introduced WebExt API)
- [X] Adds authors
- [X] Reduces requested permissions (EXIF data popup is only reachable via context menu, so don't require <all_urls>)
- [X] Sorts keys in alphabetical order and removes empty lines (all for easier editing in the future)
- [X] Increments version (still 0., because, IMHO, it's not ready to be released to the world.. yet!)

### CODE REVIEW

**Comment by gcp** [comment]
"EXIF data popup is only reachable via context menu, so don't require <all_urls>"

Do we not require it for this potentially cross-CORS call?
https://github.com/gcp/wxif/blob/56d5059fdade6af5eb205a04fbc747d122b603e2/contentscript.js#L52

File: manifest.json [hunk 1]
```diff
@@ -1,27 +1,28 @@
 {
-
-  "manifest_version": 2,
-  "name": "wxIF",
-  "version": "0.2",
+  "applications": {
```
Comment (gcp) [inline manifest.json]: Why is this needed? What's the point of it?

We don't use anything requiring fixed IDs. I don't see why we care what the installed filename is.

File: manifest.json [hunk 2]
```diff
@@ -1,27 +1,28 @@
...
+    "gecko": {
+      "id": "wxif@morbo.org",
+      "strict_min_version": "48.0"
+    }
+  },
+  "author": "Gian-Carlo Pascutto & Martin Baranski",
```
Comment (gcp) [inline manifest.json]: Not sure if adding everybody who contributes patches is doable in the long run.

File: manifest.json [hunk 3]
```diff
@@ -1,27 +1,28 @@
...
-    "contextMenus",
-    "activeTab"
+    "activeTab",
+    "contextMenus"
   ],
-
-  "background": {
-    "scripts": ["backgroundscript.js"]
-  }
+  "version": "0.3"
```
Comment (gcp) [inline manifest.json]: Let's bump this when we are ready to do a new release instead.

File: manifest.json [hunk 4]
```diff
@@ -1,27 +1,28 @@
 {
-
-  "manifest_version": 2,
-  "name": "wxIF",
-  "version": "0.2",
+  "applications": {
+    "gecko": {
+      "id": "wxif@morbo.org",
+      "strict_min_version": "48.0"
```
Comment (gcp) [inline manifest.json]: What's the point of this? What value does it add?

File: manifest.json (hunk 3 above)
Comment (Eagle3386) [reply manifest.json]: 👍

File: manifest.json (hunk 2 above)
Comment (Eagle3386) [reply manifest.json]: Sound's good to me. I'll update the file ASAP. 😃

**Comment by gcp** [comment]
>Do we not require it for this potentially cross-CORS call?

The case where I'm worried about is a webpage on bar.com with a simple 

The script will run in bar.com context and get permissions on the page, but the XmlHttpRequest will do a cross-CORS request to foo.com. I would suspect that this is not allowed without all_urls, but I'm far from an expert on that area.

File: manifest.json (hunk 4 above)
Comment (Eagle3386) [reply manifest.json]: This adds two benefits:
1. Without `strict_min_version` set, FF defaults to `42a1` which makes it (theoretically) possible to install it on non-WebExt versions, hence my PR comment about it:
   > Sets minimum version to FF48 (introduced WebExt API)
2. Adding `id`, together with a (hopefully non-existent) "mail address", enables easier recognition within FF's profile folder: the current XPI filename `{8b3ee44a-0805-4c2c-acef-15aab34fbd20}.xpi` is changed to `wxif@morbo.org.xpi` - just like other WebExtensions already do: `uBlock0@raymondhill.net.xpi`, `https-everywhere-eff-bill@eff.org.xpi`, etc.
  See MDN on [Extension ID format](https://developer.mozilla.org/en-US/Add-ons/WebExtensions/manifest.json/applications#Extension_ID_format) for further details.

File: manifest.json (hunk 1 above)
Comment (Eagle3386) [reply manifest.json]: Answered below in my comment below for `manifest.json`. It's mainly for easier finding, but also good practice, too.

File: manifest.json (hunk 3 above)
Comment (Eagle3386) [reply manifest.json]: Fixed. 😃

File: manifest.json (hunk 2 above)
Comment (Eagle3386) [reply manifest.json]: Fixed. 😃

**Comment by Eagle3386** [comment]
The [MDN states](https://developer.mozilla.org/en-US/Add-ons/WebExtensions/manifest.json/permissions#activeTab_permission) about the activeTab permission:

File: manifest.json (hunk 4 above)
Comment (Eagle3386) [reply manifest.json]: Additionally, [MDN also states](https://developer.mozilla.org/en-US/Add-ons/WebExtensions/WebExtensions_and_the_Add-on_ID#When_do_you_need_an_add-on_ID) that a constant ID is required if sync-support is desired.

For implementing #4, I'm aiming for such support (see [my comment](https://github.com/gcp/wxif/issues/4#issuecomment-359801116) there).

**Comment by gcp** [comment]
>The MDN states about the activeTab permission:

That doesn't seem to contradict what I said?

File: manifest.json (hunk 4 above)
Comment (gcp) [reply manifest.json]: >FF defaults to 42a1 which makes it (theoretically) possible to install it on non-WebExt versions,

I'm not sure what you mean by "FF defaults it" but AMO clearly recognizes and serves it to FF57+ only.

>Additionally, MDN also states that a constant ID is required if sync-support is desired.

We don't use sync support now, and syncing the position of the window seems like a bad idea (it is *not* something you want to carry over between machines!). We can add this when we have a use for it, and we do not now.

**Comment by gcp** [comment]
I think you want the extra locals in separate pull requests.

**Comment by Eagle3386** [comment]
> That doesn't seem to contradict what I said?

It does, because it states:
> extension that **wants to run a script in the current page** when the user clicks

and the documentation for `executeScript` [states](https://developer.mozilla.org/en-US/Add-ons/WebExtensions/API/tabs/executeScript) that its `details` object has a property `file` for a JavaScript content script which can then be used to handle external requests - if ever needed.

**Comment by Eagle3386** [comment]
> I think you want the extra locals in separate pull requests.

Yes, but I seem unable to do so, because GH added it, although I clicked "Pull request". 😞

**Comment by gcp** [comment]
>and the documentation for executeScript states that its details object has a property file for a JavaScript content script **which can then be used to handle external requests - if ever needed**.

I don't see where you found the highlighted part or that "handle external requests" would necessarily imply that CORS no longer applies.

I had a simple try, exactly testing the scenario I **already described above**, and wxIF no longer works if it is missing all_urls.

**Comment by gcp** [comment]
Closing this now: it got intermingled with locale updates, and I'm very unlikely to take any of the changes proposed.

**Comment by Eagle3386** [comment]
Sorry for misreading your earlier comment. I'll try to separate my (updated) PR in different branches.

File: manifest.json (hunk 4 above)
Comment (Eagle3386) [reply manifest.json]: I don't want to implement syncing of the window position, but instead set options (e. g. maps provider, display units of lat/long, etc.).

Comment (Eagle3386) [reply manifest.json]: MDN [states](https://developer.mozilla.org/en-US/Add-ons/WebExtensions/manifest.json/applications) that omitting `strict_min_version` results in AMO setting it to `42a1`:
> `strict_min_version`: minimum version of Gecko to support. Versions containing a "*" are not valid in this field. **Defaults to "42a1".**
